src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -179,35 +189,28 @@ void GcsPlacementGroupScheduler::ReserveResourceFromNode(
   RAY_LOG(INFO) << "Leasing resource from node " << node_id
                 << " for bundle: " << bundle->DebugString();
   lease_client->RequestResourceReserve(
-      *bundle, [this, node_id, bundle, callback](
+      *bundle, [node_id, bundle, callback](
                    const Status &status, const rpc::RequestResourceReserveReply &reply) {
         // TODO(AlisaWu): Add placement group cancel.
         auto result = reply.success() ? Status::OK()
                                       : Status::IOError("Failed to reserve resource");
-        auto bundles = node_to_bundles_when_leasing_.find(node_id);
-        RAY_CHECK(bundles != node_to_bundles_when_leasing_.end());
-        auto bundle_iter = bundles->second.find(bundle->BundleId());
-        RAY_CHECK(bundle_iter != bundles->second.end());
         if (result.ok()) {
           RAY_LOG(INFO) << "Finished leasing resource from " << node_id
                         << " for bundle: " << bundle->DebugString();
         } else {
           RAY_LOG(WARNING) << "Failed to lease resource from " << node_id
                            << " for bundle: " << bundle->DebugString();
         }
-        // Remove the bundle from the leasing map as the reply is returned from the
-        // remote node.
-        bundles->second.erase(bundle_iter);
-        if (bundles->second.empty()) {
-          node_to_bundles_when_leasing_.erase(bundles);
-        }
         callback(result);
       });
 }
 
 void GcsPlacementGroupScheduler::CancelResourceReserve(
     const std::shared_ptr<BundleSpecification> &bundle_spec,
     const std::shared_ptr<ray::rpc::GcsNodeInfo> &node) {
+  if (node == nullptr) {
```

Comment:
Why node will be nullptr?

Reply:
gcs_node_manager_.GetNode returns nullptr if the node is not alive anymore.

Reply:
How about add a log?

Reply:
Done.